scalameta/semantic/shared/src/main/scala/scala/meta/semantic/Attributes.scala

```diff
@@ -4,15 +4,18 @@ package semantic
 import org.scalameta.data._
 import scala.meta.inputs._
 import scala.meta.internal.inputs._
+import scala.meta.parsers.{XtensionParsersDialectInput, XtensionParseDialectInput}
 
 @data class Attributes(
   input: Input,
   dialect: Dialect,
   names: List[(Position, Symbol)],
   messages: List[Message],
   denotations: List[(Symbol, Denotation)],
-  sugars: List[(Position, String)]
+  sugars: List[(Position, Attributes)]
 ) {
+  def sugarStats: List[(Position, Stat)] = sugars.map { case (pos, attrs) => pos -> attrs.stat }
+  def stat: Stat = dialect(input).parse[Stat].get
```

Comment:
Turns out we don't! I've replaced `String` with a custom class called `Sugar` (instead of `Attributes`).